### ISSUE TITLE
Properties can't exist in more than one index

### ISSUE THREAD

**Issue by mattjohnsonpint** [mj1856/EntityFramework.IndexingExtensions#2]
``` csharp
modelBuilder.Entity<Foo>()

    .HasIndex("Index1",
        e => e.Property(x => x.A),
        e => e.Property(x => x.B))

    .HasIndex("Index2",
        e => e.Property(x => x.B),
        e => e.Property(x => x.C));
```

Field `B` should belong to both indexes.  However, `Index1` gets created without it.

**Comment by mattjohnsonpint** [comment]
This is because there can only be one `IndexAnnotation` per property, which should contain information about all indexes the property is part of.

http://stackoverflow.com/a/24346546/634824

**Comment by mattjohnsonpint** [comment]
Fixed in 10b66939e30b5d8d33d9d7fbee8cbfbfd66e0b29
